File: 백준/Bronze/1009. 분산처리/분산처리.py
import sys
num = int(input())

for i in range(num):
    a, b = map(int, sys.stdin.readline().split())
    res = a%10

    if res == 0:
        print(10)
    elif res in [1, 5, 6]:
        print(res)
    elif res in [4, 9]:
        if b % 2 == 0:
            print(res ** 2 % 10)
        else:
            print(res)
    else:
        if b % 4 == 0:
            print(res ** 4 % 10)
        else:
            print(res ** (b % 4) % 10)

File: 백준/Bronze/1009. 분산처리/분산처리.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;

public class Main {
	public static void main(String[] args) {
		try {
			BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
			
			int testCase = Integer.parseInt(br.readLine());
			
			for (int i=0; i<testCase; i++) {
				String data[] = br.readLine().split(" ");
				int A = Integer.parseInt(data[0]);
				int B = Integer.parseInt(data[1]);
				
				int res = 1;
				
				for (int j=0; j<B; j++) {
					res = (res*A)%10;
				}
                
                if(res == 0) {
                    res=10;
				}
				
				System.out.println(res);
			}
			
		}catch(IOException e) {
			System.out.println("error");
		}
	}
}